polygeist/include/mlir/Dialect/Polygeist/Utils/TransformUtils.h

```diff
@@ -136,10 +136,13 @@ class FunctionKernelInfo {
   void getKernelCallers(FunctionOpInterface func,
                         SmallVectorImpl<gpu::GPUFuncOp> &kernels) const;
 
-  /// Returns the kernel body function of \p kernel. The kernel body function is
-  /// the lambda/functor associated with the SYCL kernel construct (e.g.,
-  /// parallel_for).
-  FunctionOpInterface getKernelBodyFunc(gpu::GPUFuncOp kernel) const;
+  /// Returns the potential 'kernel body functions' of \p kernel. A 'kernel body
+  /// function' is the lambda/functor associated with the SYCL kernel construct
+  /// (e.g., parallel_for). Note that transformation passes might have cloned
+  /// the kernel body to specialize it. This function returns all possible
+  /// kernel body functions, including specializations.
+  std::set<FunctionOpInterface>
```

Comment:
Any reason for not using SmallSet ?

Reply:
I guess we could use it, but we do not know how many function bodies would be found. Should be small though so should have a SmallSet with say  N = 4.

Reply:
We are using std::set in several places so if we do this change we should be consistent and change them all to use llvm::SmallSet.

Reply:
It works like smallvector, so it is not a max number, just more efficient if you stay within your initial space. I agree, 4 is probably a good number.

It is a bit nitpicking, and as there is already many uses of `std::set` you may have to refactor quite a bit. So up to you if you want to do the change.

Reply:
Sure I will consider it. But in a separate PR.